### PR TITLE
Models related with Paper & Parser for Loading Data

### LINKED ISSUE #9: Implementing Models related with papers & Paper page
We need to implement **models** with Django, **starting from tables directly related with papers from academic DB sites**. Records of the tables should be automatically added **by parsing meta-info files** downloaded from the websites. Also on the front-end side, **paper (detail) page** should be implemented.

### PR DESCRIPTION
**Pull request by davin111**
Related Issue: #9 

# Major change
## Rename and Create some Directories
Renamed 'backend/backend_python' to 'backend/PapersFeed_backend'.
Renamed 'backend/application' to 'backend/papersfeed'.

Added a new app 'backend/papers'.

As we discussed, please respect our convention of directory structures.

## Implemented Django Models related with Papers
Implemented Django models directly related with papers downloaded from academic DB sites.
- Paper
- Author
- PaperAuthor
- Publisher
- Publication
- PaperPublication
- Reference
- Keyword
- PaperKeyword

**I created some fields different from the [original design](https://github.com/swsnu/swpp2019-team3/wiki/System-Architecture#model) because I found some types are inadequate to contain real data.**
- All Decimal -> Char (In many cases, there are also characters.)
- 'volume' and 'issue' in PaperPublication: PositiveInteger -> Char (In some cases, there are characters.)
- 'date' in PaperPublication: DateTime -> Date (Publication 'time' is never specified.)

**Because there was no actual data schema, please just reset your DB and remigrate when this PR is merged.**
You can do it in some ways as follows.

1. Log in to mysql as root or 'PapersFeed', and recreate DB.
In terminal, please type this and enter the password.
`mysql -u root -p` or `mysql -u PapersFeed -p`

2. By type this, drop the existing DB and  recreate new one.
`mysql> DROP DATABASE PapersFeed_DB;`
`mysql> CREATE DATABASE PapersFeed_DB;`

3. Migrate.
In the 'backend' dir, please type this to make migrations and migrate.
`python3 manage.py migrate`
If something is wrong with migration files, just delete it('papers/migrations/0001_initial.py') and recreate it by `python3 manage.py makemigrations`.

## Made Parser and Added example Json files
Made 'papers/utils/parser.py' which is **able to parse record files(tsv format) from [Web of Science](http://apps.webofknowledge.com).**
Added example **record files**('papers/utils/cs_10.txt' and 'papers/utils/cs_500.txt') in 'papers/utils' and corresponding **parsed json files**('papers/fixtures/cs_10.json' and 'papers/fixtures/cs_500.json'). **In Django, the json files are called fixtures.**

**I made some assumption about paper information for parsing and putting data into our database.**
- If the publication month or day is not specified but the year is, the parser assumes it is '01'.
- ...(I'll add more assumption ASAP.)

**If you want to load parsed data(json format) to PapersFeed_DB, just type this in the 'backend' dir.** 
`python3 manage.py loaddata {fixture_name}`

You can use 'cs_10' or 'cs_500' as the fixture_name. The each trailing number is the number of records of papers.

If you want your data in MySQL, follow below after logging in.
`mysql> USE PapersFeed_DB;`

You can see all tables.
`mysql> SHOW TABLES;`

Also, you can retrieve some fields(columns) from each table.
스크린샷 2019-10-24 05 36 25

## Made Travis CI ensure Test Coverage Threshold
I made Travis CI ensure **test coverage of frontend and backend to be greater than or equal to 90%**.

## Made Travis CI prepare MySQL env for Backend
Now backend parts of our project are built correctly.

# Minor change
Added a '.gitignore' file in root dir.
Deleted '.idea' dir and the included files in root dir.

Add a trivial unit test to Main.test.js for test coverage. But I think the implementation of Main.js related with it could be improved using enum.

Some backend codes are commented because of test coverage or pylint.
In some parts of code, pylint 'disable' and 'enable' is added.
Tentatively, I turned off **fixme** and **duplicate-code** checkers of pylint.

### CODE REVIEW

File: backend/papers/models/author.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+from django.db import models
+
+from .paper import Paper
+
+
+class Author(models.Model):
+    """Author Model"""
+    first_name = models.CharField(max_length=50)
+    last_name = models.CharField(max_length=50)
+    email = models.EmailField()
```
Comment (kosumin) [inline backend/papers/models/author.py]: How about adding max_length to this field?

Comment (davin111) [reply backend/papers/models/author.py]: I added to it! Thank you :)

Comment (davin111) [reply backend/papers/models/author.py]: By the way, it seems that default max_length of the field is 254 when not specified.

Comment (kosumin) [reply backend/papers/models/author.py]: Oh, that's good. Thank you for the information. Then I will resolve this conversation.

**Comment by davin111** [comment]
> Looks good to me. So I guess we can now use an api kind of 'axios.get('/paper/')'?

@njsh4261 
Maybe not. Backend APIs should be implemented by @abcthing. I only created models. But you can play with these models and test records in `python3 manage.py shell`.

**Pull request merged**